src/backend/base/langflow/components/data/URL.py

```diff
@@ -71,7 +72,9 @@ def ensure_url(self, string: str) -> str:
         return string
 
     def fetch_content(self) -> list[Data]:
-        urls = [self.ensure_url(url.strip()) for url in self.urls if url.strip()]
+        # check if the urls are list or not
+        urls_list = [self.urls] if isinstance(self.urls, str) else self.urls
```

Comment:
If `self.urls` is `None`, does the below behavior still work? Or do we need to add an explicit `None` check and return an error?

Reply:
I'd say a check could be useful to catch unexpected errors. This should never happen but will help us avoid regression.

Reply:
I agree with Jordan. I'd say a check could be useful to catch unexpected errors. This should never happen but will help us avoid regression.